documentation/modules/oauth/con-oauth-reauthentication.adoc

```diff
@@ -1,21 +1,40 @@
 // Module included in the following assemblies:
 //
-// assembly-oauth.adoc
+// assembly-oauth-authentication.adoc
 
 [id='con-oauth-reauthentication-{context}']
 = Session re-authentication for Kafka brokers
 
-The Kafka _SASL OAUTHBEARER_ mechanism, which is used for OAuth 2.0 authentication in Strimzi, supports a Kafka feature called the _re-authentication_ mechanism.
+OAuth 2.0 authentication in Strimzi supports a Kafka feature called _session re-authentication_. 
+To enable and configure session re-authentication, you set a time value for `maxSecondsWithoutReauthentication` in the `oauth` listener configuration for the Kafka broker.
 
-When the re-authentication mechanism is enabled in the configuration of an `oauth` type listener, the broker's authenticated session expires when the access token expires.
-The client must then re-authenticate to the existing session by sending a new, valid access token to the broker, without dropping the connection.
+Session re-authentication can be used with _fast local JWT_ or _introspection endpoint_ token validation.
+
+There are some differences in how session expiry is handled in SASL_OAUTHBEARER and SASL_PLAIN authentication.
+
+.Session expiry with OAUTHBEARER
+
+When session re-authentication is configured for SASL_OAUTHBEARER, the broker's authenticated session will expire at the configured `maxSecondsWithoutReauthentication`. 
+The session will expire earlier if the access token expires before the `maxSecondsWithoutReauthentication`. 
+The duration of the access token always take precedence over `maxSecondsWithoutReauthentication`.
+
+.Session expiry with PLAIN
+
+When session re-authentication is configured with SASL_PLAIN, the broker's authenticated session will expire at the configured `maxSecondsWithoutReauthentication`. 
+If `maxSecondsWithoutReauthentication` is not configured, SASL_PLAIN clients can remain connected to brokers indefinitely, without needing to re-authenticate.
+
+.Client re-authentication
+
+When the broker's authenticated session expires, the client must re-authenticate to the existing session by sending a new, valid access token to the broker, without dropping the connection.
```

Comment:
This section seems a bit disconnected. Does it relate to _Session expiry with PLAIN_? Or both _Session expiry with PLAIN_ and _Session expiry with OAUTHBEARER_?

Reply:
I think this part applies to both OAUTHBEARER and PLAIN.

 @mstruk - Please can you read the updated doc for Session re-authentication and let me know if anything looks out of place.

Do we need more information about _client re-authentication_ when using PLAIN?

Reply:
Re-authentication is related to the enforcement of session expiry - it allows a new session authentication from the client, effectively the start of a new session by reusing existing connection rather than dropping it. The mechanics is really in the Kafka Client library, not something the client has to explicitly take care of in the code. I see this chapter more as an explanation what's going on in the background for user to better understand the mechanics.

Reply:
Great info, thanks. I updated the module again based on this text.